### PR TITLE
Add exit_early argument to ReplayManager.run()

### PR DESCRIPTION
**Pull request by tysmith**

### CODE REVIEW

File: grizzly/replay/test_replay.py [hunk 1]
```diff
@@ -163,6 +163,29 @@ def test_replay_07(mocker, tmp_path):
...
+    target.detect_failure.side_effect = None
+    with ReplayManager([], server, target, use_harness=False) as replay:
+        assert not replay.run(testcases, repeat=4, min_results=4, exit_early=False)
+        assert target.close.call_count == 1
+        assert replay.status.iteration == 4
+        assert replay.status.results == 0
+        assert replay.status.ignored == 0
+    target.reset_mock()
+    # ignore early success (perform all repeats)
+    target.reset_mock()
```
Comment (jschwartzentruber) [inline grizzly/replay/test_replay.py:179]: back to back `reset_mock()` calls

File: grizzly/replay/replay.py [hunk 2]
```diff
@@ -104,14 +104,19 @@ def report_to_filesystem(path, results, tests=None):
...
+    def run(self, testcases, repeat=1, min_results=1, exit_early=True, idle_delay=0, idle_threshold=0):
         """Run testcase replay.
 
         Args:
             testcases (list): One or more TestCases to run.
             repeat (int): Maximum number of times to run the TestCase.
             min_results (int): Minimum number of results needed before run can
                                be considered successful.
+            exit_early (bool): If True the minimum required number of iterations
+                               are perform to either meet `min_results` or
```
Comment (jschwartzentruber) [inline grizzly/replay/replay.py:116]: perform -> performed

**Review by jschwartzentruber — APPROVED** [review]

File: grizzly/replay/test_replay.py (hunk 1 above)
Comment (tysmith) [reply grizzly/replay/test_replay.py]: Good catch.

File: grizzly/replay/replay.py (hunk 2 above)
Comment (tysmith) [reply grizzly/replay/replay.py]: Fixed.

**Pull request merged**